src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -593,14 +594,74 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
                             Slots.RUBRIC_ROW_BODY_FRAGMENTS, tableBodyFragmentHtml.toString());
             tableBodyHtml.append(tableRow).append(Const.EOL);
         }
+
+        String statsTitle = "Response Summary";
+
+        if ("student".equals(view)) {
+            if (responses.size() == responsesForStatistics.size()) {
+                statsTitle = "Response Summary (of visible responses)";
+            } else {
+                statsTitle = "Response Summary (of received responses)";
+            }
+        }
         
         
         return Templates.populateTemplate(
                 FormTemplates.RUBRIC_RESULT_STATS,
-                Slots.STATS_TITLE, "student".equals(view) ? "Response Summary (of visible responses)" : "Response Summary",
+                Slots.STATS_TITLE, statsTitle,
                 Slots.TABLE_HEADER_ROW_FRAGMENT_HTML, tableHeaderFragmentHtml.toString(),
                 Slots.TABLE_BODY_HTML, tableBodyHtml.toString());
     }
+
+    /**
+     * Returns a list of FeedbackResponseAttributes filtered according to view, question recipient type
+     * for the Statistics Table
+     */
+    private List<FeedbackResponseAttributes> filterResponsesForStatistics(List<FeedbackResponseAttributes> responses,
+            FeedbackQuestionAttributes question, String studentEmail,
+            FeedbackSessionResultsBundle bundle, String view) {
+
+
+        boolean isViewedByStudent = "student".equals(view);
+        if (!isViewedByStudent) {
+            return responses;
+        }
+
+        FeedbackParticipantType[] typesToSkipFiltering = {FeedbackParticipantType.INSTRUCTORS,
+                FeedbackParticipantType.NONE, FeedbackParticipantType.SELF};
+        FeedbackParticipantType recipientType = question.getRecipientType();
+
+        for (FeedbackParticipantType skipType : typesToSkipFiltering) {
+            if (recipientType.equals(skipType)) {
+                return responses;
+            }
+        }
+
+        FeedbackParticipantType[] typesToFilterByTeams = {FeedbackParticipantType.OWN_TEAM,
+                FeedbackParticipantType.TEAMS};
+        boolean toFilterByTeams = false;
+
+        for (FeedbackParticipantType teamType : typesToFilterByTeams) {
+            if (recipientType.equals(teamType)) {
+                toFilterByTeams = true;
+            }
+        }
+
+        List<FeedbackResponseAttributes> receivedResponses = new ArrayList<>();
+        String recipientString = toFilterByTeams ? bundle.getTeamNameForEmail(studentEmail) : studentEmail;
+
+        for (FeedbackResponseAttributes response : responses) {
+            boolean isReceivedResponse = response.recipient.equals(recipientString);
+            if (isReceivedResponse) {
+                receivedResponses.add(response);
+            }
+        }
+        if (receivedResponses.isEmpty()) {
```

Comment:
For rubric questions where students are recipients, if a student receives responses he is shown only the stats of responses that he received, otherwise if he did not receive any response, he will be shown the stats of all visible responses. I was wondering if this behaviour was alright, or should we not show the stats of visible responses if the student did not receive any responses. 

(This to applies to teams as well)